From: KonTy <konty@users.noreply.github.com>
Date: Fri, 10 Apr 2026 12:00:00 +0200
Subject: [PATCH] Qt 5.12 compatibility fixes

Compatibility fixes for Qt 5.12 instead of newer ones.

Upstream-Status: Pending
Signed-off-by: KonTy <konty@users.noreply.github.com>
---
diff --git i/CMakeLists.txt w/CMakeLists.txt
index 1d4734e3a..64dd85b72 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -11,7 +11,7 @@ feature_summary(WHAT REQUIRED_PACKAGES_NOT_FOUND FATAL_ON_MISSING_REQUIRED_PACKA
 
 set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH} ${CMAKE_SOURCE_DIR}/cmake)
 
-set(REQUIRED_QT_VERSION 5.15.2)
+set(REQUIRED_QT_VERSION 5.12.0)
 
 include(KDEInstallDirs)
 include(KDECMakeSettings)
@@ -71,7 +71,7 @@ configure_package_config_file(
 )
 
 
-add_definitions(-DQT_DISABLE_DEPRECATED_BEFORE=0x050f02)
+add_definitions(-DQT_DISABLE_DEPRECATED_BEFORE=0x050c00)
 
 
 ########### Targets ###########
diff --git i/src/attendee.h w/src/attendee.h
index 420699b13..de9278dfd 100644
--- i/src/attendee.h
+++ w/src/attendee.h
@@ -17,6 +17,7 @@
 #define KCALCORE_ATTENDEE_H
 
 #include <QMetaType>
+#include <QVector>
 #include <QSharedDataPointer>
 
 #include "customproperties.h"
diff --git i/src/calendar.h w/src/calendar.h
index 53ebcb5f1..b313e296f 100644
--- i/src/calendar.h
+++ w/src/calendar.h
@@ -50,7 +50,7 @@ This API needs serious cleaning up:
 /** Namespace for all KCalendarCore types. */
 namespace KCalendarCore
 {
-Q_NAMESPACE_EXPORT(KCALENDARCORE_EXPORT)
+Q_NAMESPACE
 
 class CalFilter;
 class Person;
diff --git i/src/icalformat_p.cpp w/src/icalformat_p.cpp
index a21a2544b..39b73bf26 100644
--- i/src/icalformat_p.cpp
+++ w/src/icalformat_p.cpp
@@ -1714,7 +1714,7 @@ void ICalFormatImpl::readIncidence(icalcomponent *parent, const Incidence::Ptr &
             // We can't change that -- in order to retain backwards compatibility.
             text = icalproperty_get_categories(p);
             const QString val = QString::fromUtf8(text);
-            const QStringList lstVal = val.split(QLatin1Char(','), Qt::SkipEmptyParts);
+            const QStringList lstVal = val.split(QLatin1Char(','), QString::SkipEmptyParts);
             for (const QString &cat : lstVal) {
                 // ensure no duplicates
                 if (!categories.contains(cat)) {
diff --git i/src/incidencebase_p.h w/src/incidencebase_p.h
index 56ad5252b..3be5a5618 100644
--- i/src/incidencebase_p.h
+++ w/src/incidencebase_p.h
@@ -13,6 +13,7 @@
 #define KCALCORE_INCIDENCEBASE_P_H
 
 #include "incidencebase.h"      // For IncidenceBase::Field
+#include "attendee.h"
 
 namespace KCalendarCore
 {
